packages/@glimmer/manager/lib/internal/defaults.ts

```diff
@@ -0,0 +1,42 @@
+import { buildCapabilities } from '../util/capabilities';
+
+import type { CapturedArguments as Arguments, HelperCapabilities } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [...Args['positional'], Args['named']]
+  | [...Args['positional']];
+
+interface FunctionHelperState<Args extends Arguments = Arguments> {
+  fn: <Return>(...args: FnArgs<Args>) => Return;
+  args: Args;
+}
+
+export class FunctionHelperManager<State extends FunctionHelperState> {
```

Comment:
@chriskrycho @dfreeman does Glint consume these types in any way? If not,  have some doubts about the utility of having this generic here. Inside the VM, we never know what the types of the user supplied objects are so things just get filled with `unknown`. If this is not useful to Glint either I think we can just remove them to keep things simple. It would be different if this is subclassable.

Reply:
>  utility of having this generic here

it was mostly for me to type less.
Happy with whatever makes the most sense though

Reply:
Assuming the Glint team has no use for it, you can probably just do:

```
type AnyFunction = (...args: unknown[]) => unknown;

interface FunctionHelperState {
  fn: AnyFunction;
  args: Argument;
}
```

You can keep keep using `State['fn']` and `State['args']` if you prefer, or you can just replace them with `AnyFunction` and `Arguments` at that point

Reply:
I’ll let Dan confirm but I don’t think we’d end up using these. I will note though that the type should probably be `(...args: any[]) => unknown` because otherwise subtyping is impossible due to the way that variance works with function parameters.

Reply:
Glint has no awareness of the managers that back template-invokable entities (as the effective interface of the entity in a template isn't robustly derivable from the type of its manager), so keeping the types simple here seems like a good call to me 👍

Reply:
latest push has simplifications. thanks all!